### PR TITLE
DevWs for Current Products Corp containing containing Current Products E-Wand

### PR DESCRIPTION
**Pull request by currentDev**

### CODE REVIEW

**Comment by tjk6** [comment]
I was wondering what the time frame may be on reviewing this and if there was anything we could do to help speed along the process.

File: devicetypes/instantpeace33473/current-products-e-wand.src/current-products-e-wand.groovy [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ */
+metadata {
+	definition (name: "Current Products E-Wand", namespace: "instantpeace33473", author: "Tyler Kass", mnmn: "SmartThingsCommunity", vid: "5f006ef2-6b52-3357-bc06-615dbce3701a") {
+        capability "instantpeace33473.tilt"
+        
+		fingerprint profileId: "0104", inClusters: "0000, 0001, 0003, 0004, 0005, 0006, 0008, 0020, 0102, 0B05, FC10", outClusters: "0003, 0019", manufacturer: "Current Products Corp", model: "E-Wand", deviceJoinName: "CPC E-Wand"
```
Comment (dkirker) [inline devicetypes/instantpeace33473/current-products-e-wand.src/current-products-e-wand.groovy:19]: You should be able to simplify this to just:

```
fingerprint manufacturer: "Current Products Corp", model: "E-Wand", deviceJoinName: "CPC E-Wand"
```

File: devicetypes/instantpeace33473/current-products-e-wand.src/current-products-e-wand.groovy [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+//For later use
+private getCLUSTER_WINDOW_COVERING() { 0xFC10 }
+private getATTRIBUTE_POSITION_TILT() { 0x26 }
+private getADDITIONAL_PARAMS() { [mfgCode:0x1263] }
+
+// parse events into attributes
+def parse(String description) {
+	log.trace "Parsing '${description}'"
+
+	String[] descArr = description.split(" ", 0)
```
Comment (dkirker) [inline devicetypes/instantpeace33473/current-products-e-wand.src/current-products-e-wand.groovy:35]: There is actually a smoother method to parse and handle the description. There are a couple of methods in the `zigbee` object that can assist. Here is an example: https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/smartsense-multi-sensor.src/smartsense-multi-sensor.groovy#L131

File: devicetypes/instantpeace33473/current-products-e-wand.src/current-products-e-wand.groovy [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    if("51".equals(commandType)) { // 0x51 signifies the packet is of type movement command
+    	String payload = descArr[len-1]
+        String position = payload.substring(2,4)
+        int posInt = zigbee.convertHexToInt(position)
+        if(posInt > 100) {
+            posInt = negateHex(posInt) // Gets the negation of the hex value to get the negative int it represents
+            if (posInt < -100) { posInt = -100 } //Cap value here
+        	//log.debug "DEBUG: Setting tiltAngle to ${posInt}"
+	    	//TEST: Make sure that this is "tiltAngle" and not "Tilt Angle"
+        	//sendEvent(name: "Tilt Angle", value: posInt)
```
Comment (dkirker) [inline devicetypes/instantpeace33473/current-products-e-wand.src/current-products-e-wand.groovy:54]: Please go ahead and remove unused commented code.

File: devicetypes/instantpeace33473/current-products-e-wand.src/current-products-e-wand.groovy (hunk 2 above)
Comment (dkirker) [reply devicetypes/instantpeace33473/current-products-e-wand.src/current-products-e-wand.groovy:35]: Another good example is https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/zigbee-thermostat.src/zigbee-thermostat.groovy#L140 which has a good use of comparing clusters and attrs/commands with constants: https://github.com/SmartThingsCommunity/SmartThingsPublic/blob/master/devicetypes/smartthings/zigbee-thermostat.src/zigbee-thermostat.groovy#L524

**Review by dkirker — COMMENTED** [review]
@currentDev Apologies for the delay. Some feedback is included inline. One other recommendation is to use tabs for consistent indenting.

**Comment by k-morawski** [comment]
Dear @tjk6 , please let us know if you are still interested to integrate your device with ST. Many thanks

**Comment by k-morawski** [comment]
Dear @currentDev and @tjk6 , could you please check @dkirker review feedback? Many thanks

**Comment by k-morawski** [comment]
> Dear @currentDev and @tjk6 , could you please check @dkirker review feedback? Many thanks

Dear @currentDev and @tjk6 , are you still interested to integrate your devices with ST platform? Negative answer from you or no answer till next Thursday would cause that I will close this integration request. Looking forward to hearing from you. Thanks

**Comment by currentDev** [comment]
Terribly sorry for the delay, we're in the midst of team replacement and have fallen back on driver progress. We are still interested in integration and will be looking at this soon.

**Comment by k-morawski** [comment]
> Terribly sorry for the delay, we're in the midst of team replacement and have fallen back on driver progress. We are still interested in integration and will be looking at this soon.

Thanks @currentDev

**Comment by k-morawski** [comment]
> > Terribly sorry for the delay, we're in the midst of team replacement and have fallen back on driver progress. We are still interested in integration and will be looking at this soon.
> 
> Thanks @currentDev

Dear @currentDev , this is just a friendly reminder, could you please continue the work on this integration?

**Comment by k-morawski** [comment]
Dear @currentDev, as there is no status change for a long time please let me close this PR and connected integration request. Please open a new request when you will be ready. Many thanks, Kamil

**Pull request closed without merging**